pkg/kubelet/cm/types.go

```diff
@@ -57,9 +62,38 @@ type CgroupManager interface {
 	Exists(string) bool
 }
 
-// QOSContainersInfo hold the names of containers per qos
+// QOSContainersInfo stores the names of containers per qos
 type QOSContainersInfo struct {
 	Guaranteed string
 	BestEffort string
 	Burstable  string
 }
+
+// PodContainerManager stores and manages pod level containers
+// The Pod workers interact with the PodContainerManager to create and destroy
+// containers for the pod.
+type PodContainerManager interface {
+	// getPodContainerName returns the pod container's absolute name
+	GetPodContainerName(*api.Pod) string
+
+	// EnsureExists takes a pod as argument and makes sure that
+	// pod cgroup exists if qos cgroup hierarchy flag is enabled.
+	// If the pod cgroup doesen't already exist this method creates it.
+	EnsureExists(*api.Pod) error
+
+	Exists(*api.Pod) bool
```

Comment:
The EnsureExists() and Exists() would be called as outlined in the next PR:

PTAL: 
https://github.com/kubernetes/kubernetes/pull/29049/commits/bdc300683bd732db6599fc507e1ebe8661a44ec2#diff-bf28da68f62a8df6e99e447c4351122dR1797